app/Models/Product.php

```diff
@@ -15,15 +17,41 @@ class Product extends Model
 
     public function category(): \Illuminate\Database\Eloquent\Relations\BelongsTo
     {
-    	return $this->belongsTo(Category::class,'category_id','id');
+        return $this->belongsTo(Category::class, 'category_id', 'id');
     }
 
 
     public function brand(): \Illuminate\Database\Eloquent\Relations\BelongsTo
     {
-    	return $this->belongsTo(Brand::class,'brand_id','id');
+        return $this->belongsTo(Brand::class, 'brand_id', 'id');
     }
 
+    /**
+     * Get the user's first name.
+     *
+     * @param string $value
+     * @return string
+     */
+    public function getUserOrderDateAttribute($value)
+    {
+        $userId = request()->input('user_id');
```

Comment:
we need to move all this block of code into two helper and repository 
one repository for line 43 to 48 

another service for get user id from request then call repo then return order item to this accessor